src/services/services.js

```diff
@@ -49,23 +141,169 @@ export const getUserById = async (id) => {
     });
 };
 
-export const getAllMembers = async () => {
-  const docRef = await db.collection('data').doc('users');
+export const getRandomId = () => `${Math.random().toString(36).substr(2, 9)}`;
+
+export const setTaskData = async (data) => {
+  db.collection('data')
+    .doc('tasks')
+    .set(
+      {
+        [data.taskId]: data,
+      },
+      { merge: true },
+    )
+    .then(() => {
+      console.log('Document successfully written!');
+    })
+    .catch((error) => {
+      console.error('Error writing document: ', error);
+    });
+};
+
+export const setUserSubTaskData = async (userId, taskId, data) => {
+  db.collection('data')
+    .doc('memberTasks')
+    .set(
+      {
+        [userId]: { [taskId]: { [data.subTaskId]: data } },
+      },
+      { merge: true },
+    )
+    .then(() => {
+      console.log('Document successfully written!');
```

Comment:
create helper. avoid duplication

Reply:
thx, icreated helper method.
```suggestion
      successfullyWrittenMessage();
```